jetbrains-core/src/software/aws/toolkits/jetbrains/services/cloudwatch/logs/actions/OpenLogGroups.kt

```diff
@@ -5,13 +5,21 @@ package software.aws.toolkits.jetbrains.services.cloudwatch.logs.actions
 
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.project.DumbAware
+import com.intellij.openapi.project.Project
+import kotlinx.coroutines.launch
 import software.aws.toolkits.jetbrains.core.explorer.actions.SingleResourceNodeAction
+import software.aws.toolkits.jetbrains.services.cloudwatch.logs.CloudWatchLogWindow
 import software.aws.toolkits.jetbrains.services.cloudwatch.logs.CloudWatchLogsNode
+import software.aws.toolkits.jetbrains.utils.ApplicationThreadPoolScope
 import software.aws.toolkits.resources.message
 
-class OpenLogGroups : SingleResourceNodeAction<CloudWatchLogsNode>(message("cloudwatch.log.open")), DumbAware {
+class OpenLogGroup : SingleResourceNodeAction<CloudWatchLogsNode>(message("cloudwatch.logs.open")), DumbAware {
     override fun actionPerformed(selected: CloudWatchLogsNode, e: AnActionEvent) {
         // TODO subsequent PRs will fill this in
         println("this will open the window")
     }
 }
+
+fun openLogGroup(project: Project, logGroupName: String) = ApplicationThreadPoolScope("openLogGroup").launch {
```

Comment:
Should we follow the pattern of delegating this class to the scope?

`OpenLogGroup : CoroutineScope by ApplicationThreadPoolScope("openLogGroup")`

Reply:
This isn't a class, it's a top level function. We could make it an object, but all the object is doing would be holding this function which is fine as well

Reply:
It looks a bit silly but making the companion object extend coroutine scope seems like it makes the most sense in this case since we need to call it from another place and there's not really a point to make a whole new object

Reply:
Oh I see now, I was confused by `actionPerformed` and thought that would call this function.

So are you sharing the window-launching logic between this Action and elsewhere? Does this actually belong with the Action? Or should it be factored out?

It looks like you could move this logic into `CloudWatchLogWindow`. Then callers of this function don't have to depend on the Action to open the window.

I'm curious if Action is a concept that exists outside the scope of performing a "standalone Action". Does the framework consider the opening of a panel from another panel to be an Action?

Reply:
An action is just a convenient representation of something that can be done by the IDE with a name and maybe an icon and description. CloudWatchLogWindow is managed by the IDE's service manager so I'm not sure if that would even work properly

Reply:
Components registered through the registration system have no issues, moved the coroutines stuff to there